### ISSUE TITLE
Validation: Check for deprecated licenses

### ISSUE THREAD

**Issue by Alexander-Wilms** [ximion/appstream#574]
Another check that could be added to `appstreamcli validate` is whether the `metadata_license` and `project_license` identifiers are deprected using the `isDeprecatedLicenseId` field in the [licenses.json](https://github.com/spdx/license-list-data/blob/main/json/licenses.json).

**Comment by ximion** [comment]
Yeah, SPDX has changed this so many times though that I am a bit burned by having to play catch-up with license name changes, and I don't want to inflict the rename dance onto others as well.

That said, SPDX has been stable the past years, so I wouldn't oppose if some INFO severity hint was added for this :-)
